pkgs/native_toolchain_c/test/clinker/treeshake_helper.dart

```diff
@@ -53,75 +59,86 @@ void runTreeshakeTests(
 
   late Map<String, int> sizes;
   sizes = <String, int>{};
-  for (final architecture in architectures) {
-    for (final clinker in [
-      (name: 'manual', linker: linkerManual),
-      (name: 'auto', linker: linkerAuto),
-      (name: 'autoEmpty', linker: linkerAutoEmpty),
-    ]) {
-      test('link test with CLinker ${clinker.name} and target '
-          '$architecture for targetOS $targetOS', () async {
-        final tempUri = await tempDirForTest();
-        final tempUri2 = await tempDirForTest();
-        final testArchive = await buildTestArchive(
-          tempUri,
-          tempUri2,
-          targetOS,
-          architecture,
-          androidTargetNdkApi: androidTargetNdkApi,
-          macOSTargetVersion: macOSTargetVersion,
-        );
+  for (final clinker in [
+    (name: 'manual', linker: linkerManual),
+    (name: 'auto', linker: linkerAuto),
+    (name: 'autoEmpty', linker: linkerAutoEmpty),
+  ]) {
+    test('link test with CLinker ${clinker.name}', () async {
+      final tempUri = await tempDirForTest();
+      final tempUri2 = await tempDirForTest();
+      final testArchive = await buildTestArchive(
+        tempUri,
+        tempUri2,
+        targetOS,
+        targetArchitecture,
+        androidTargetNdkApi: androidTargetNdkApi,
+        macOSTargetVersion: macOSTargetVersion,
+        iOSTargetVersion: iOSTargetVersion,
+        iOSTargetSdk: iOSTargetSdk,
+      );
 
-        final linkInputBuilder = LinkInputBuilder()
-          ..setupShared(
-            packageName: 'testpackage',
-            packageRoot: tempUri,
-            outputFile: tempUri.resolve('output.json'),
-            outputDirectoryShared: tempUri2,
-          )
-          ..setupLink(assets: [], recordedUsesFile: null)
-          ..addExtension(
-            CodeAssetExtension(
-              targetOS: targetOS,
-              targetArchitecture: architecture,
-              linkModePreference: LinkModePreference.dynamic,
-              cCompiler: cCompiler,
-              android: androidTargetNdkApi != null
-                  ? AndroidCodeConfig(targetNdkApi: androidTargetNdkApi)
-                  : null,
-              macOS: macOSTargetVersion != null
-                  ? MacOSCodeConfig(targetVersion: macOSTargetVersion)
-                  : null,
-            ),
-          );
+      final linkInputBuilder = LinkInputBuilder()
+        ..setupShared(
+          packageName: 'testpackage',
+          packageRoot: tempUri,
+          outputFile: tempUri.resolve('output.json'),
+          outputDirectoryShared: tempUri2,
+        )
+        ..setupLink(assets: [], recordedUsesFile: null)
+        ..addExtension(
+          CodeAssetExtension(
+            targetOS: targetOS,
+            targetArchitecture: targetArchitecture,
+            linkModePreference: LinkModePreference.dynamic,
+            cCompiler: cCompiler,
+            android: androidTargetNdkApi != null
```

Comment:
should this be `targetOs == OS.android` instead, with a `!` on `androidTargetNdkApi`? Ditto for the other OSes.

Reply:
Since we check at the top of the method that androidTargetNdkApi must be set iff targetOS is Android it shouldn't matter logically. I find the current way without the extra null check easier to read, which is why I opted for that one.